File: CourseProjects/CourseProjects/VisualStudioProjects/Agile/MockObjectsGenerator/TestFixture/Math.cs
using System;

namespace TestFixture {
	public abstract class Math {
		public abstract int add(int no1, int no2);
		public abstract int multiply(int no1, int no2);
		public abstract double divide(int no1, int no2);
		public abstract void accumulate(int num);
		public abstract int accumulateTotal();

	}
}

File: CourseProjects/CourseProjects/JavaEclipseProjects/UnitTestingExamples/src/demos/tdd/testng/basic/Math.java
package demos.tdd.testng.basic;

public class Math {
	public Math(int no1, int no2) {
		if (no1 == 0 || no2 == 0) {
			throw new IllegalArgumentException("Zero not allowed");
		}
		this.no1 = no1;
		this.no2 = no2;
	}
	public int add() {
		return no1 + no2;
	}
	public int subtract() {
		return no1 - no2;
	}
	public int multiply() {
		return no1 * no2;
	}
	private int no1;
	private int no2;
}